packages/sources/layer2-sequencer-health/src/endpoint/health.ts

```diff
@@ -133,15 +72,15 @@ export const execute: ExecuteWithConfig<ExtendedConfig> = async (request, _, con
   // #2 Option: Check block height
   // If every method succeeds, the Network is considered healthy
   // If any method fails, an empty tx is sent. This determines the final state
-  const wrappedMethods = [getSequencerHealth, getL2NetworkStatus].map(_tryMethod)
+  const wrappedMethods = [getSequencerHealth, checkIsNetworkProgressing].map(_tryMethod)
   for (let i = 0; i < wrappedMethods.length; i++) {
     const method = wrappedMethods[i]
-    const isHealthy = await method(network, config.delta, config.deltaBlocks)
+    const isHealthy = await method(network, config)
     if (!isHealthy) {
       Logger.info(`Checking unhealthy network ${network} with transaction submission`)
       let isHealthyByTransaction
       try {
-        isHealthyByTransaction = await getStatusByTransaction(network, config.timeoutLimit)
+        isHealthyByTransaction = await getStatusByTransaction(network, config)
       } catch (e: any) {
         throw new AdapterDataProviderError({
           network,
```

Comment:
Right now we just respond with `ok` and we don't do the check again.  We can definitely improve the workflow but I think that this is a larger change as it would impact the flow for all other networks.  Can we create a separate ticket and discuss this for the future?